src/app/beer_garden/config.py

```diff
@@ -66,21 +74,20 @@ def generate(args: List[str]):
         YapconfLoadError: Missing 'config' configuration option (file location)
     """
     spec = YapconfSpec(_SPECIFICATION, env_prefix="BG_")
-    config = _generate_config(spec, args)
 
-    # Bootstrap items shouldn't be in the generated config file
-    # We mimic a migration as it's the easiest way to filter out bootstrap items
-    items = [item for item in spec._yapconf_items.values() if not item.bootstrap]
-    filtered_config = {}
-    for item in items:
-        item.migrate_config(
-            config, filtered_config, always_update=False, update_defaults=False
-        )
+    parser = ArgumentParser()
+    spec.add_arguments(parser)
+    args = parser.parse_args(args)
+
+    bootstrap = spec.load_filtered_config(vars(args), "ENVIRONMENT", bootstrap=True)
+    config = spec.load_filtered_config(
+        vars(args), "ENVIRONMENT", exclude_bootstrap=True
+    )
 
     dump_data(
-        filtered_config,
-        filename=config.configuration.file,
-        file_type=_get_config_type(config),
+        config,
+        filename=bootstrap.configuration.file,
+        file_type=_get_config_type(bootstrap),
```

Comment:
do we still want to support `json` and `yaml`? I'm fine saying, our config files are yaml.

Reply:
I'm good with that. Even easier :smile: